### PR TITLE
Adding new solver for coloring problem based on cost function network

### PR DESCRIPTION
**Pull request by g-poveda**
- modeling of coloring problem as a cfn (see doc here https://toulbar2.github.io/toulbar2/userdoc.html)
- toulbar2 not added as hard dependency because not working on some platform
- example script running fine for platform where pytoulbar2 is provided.

### CODE REVIEW

File: discrete_optimization/coloring/solvers/greedy_coloring.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def __init__(
             params_objective_function=params_objective_function,
         )
 
-    def solve(self, **kwargs):
+    def solve(self, **kwargs) -> ResultStorage:
```
Comment (nhuet) [inline discrete_optimization/coloring/solvers/greedy_coloring.py:75]: Should have been done in another PR ;p  (and for some reason, this broke mypy ?!)

File: discrete_optimization/coloring/solvers/coloring_toulbar_solver.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    ):
+        self.problem = problem
+        (
+            self.aggreg_sol,
+            self.aggreg_from_dict_values,
+            self.params_objective_function,
+        ) = build_aggreg_function_and_params_objective(
+            self.problem, params_objective_function=params_objective_function
+        )
+        self.model: pytoulbar2.CFN = None
```
Comment (nhuet) [inline discrete_optimization/coloring/solvers/coloring_toulbar_solver.py:46]: This is not correct typing. Integrate my commits on nht/toulbar_coloring to be mypy compliant.

**Review by nhuet — COMMENTED** [review]
New feature ok. I made it pass mypy in my branch nht/toulbar_coloring. You can take my commits there.

File: discrete_optimization/coloring/solvers/coloring_toulbar_solver.py (hunk 2 above)
Comment (g-poveda) [reply discrete_optimization/coloring/solvers/coloring_toulbar_solver.py]: done (normally ?!)

**Review by nhuet — APPROVED** [review]
